src/index.js

```diff
@@ -264,6 +275,33 @@ class IpfsRepo {
     }
   }
 
+  async _isAutoMigrationEnabled () {
+    let autoMigrateConfig
+    try {
+      autoMigrateConfig = await this.config.get(AUTO_MIGRATE_CONFIG_KEY)
+    } catch (e) {
+      if (e.code === ERRORS.NotFoundError.code) {
+        autoMigrateConfig = true // Config's default value is True
+      } else {
+        throw e
+      }
+    }
+
+    return autoMigrateConfig && this.options.autoMigrate
```

Comment:
I think this is not deciding if it should migrate correctly! Or at least not consistently with how we usually override config.

1. `true` in config and `true` in options = `true` ✅
1. `true` in config and `false` in options = `false` ✅
1. `true` in config and `undefined` in options = `undefined` ❌ (expected `true`)
1. `false` in config and `true` in options = `false` ❌ (expected `true`)
1. `false` in config and `false` in options = `false` ✅
1. `false` in config and `undefined` in options = `false` ✅
1. `undefined` in config and `true` in options = `undefined` ❌ (expected `true`)
1. `undefined` in config and `false` in options = `undefined` ✅
1. `undefined` in config and `undefined` in options = `undefined` ❌ (expected `true`)

For 3, 6, and 8: in the code `this.options.autoMigrate` will normally be `true`/`false` unless the user passes `{ autoMigrate: undefined }`.

The behaviour I was expecting was:

If the user explicitly passed an option to `autoMigrate` as `true`/`false` then that is the value that should be used. If not explicitly passed as an option then we should check for a value in repo config otherwise we use the default (true).

The problem comes when the user doesn't pass a value for `autoMigrate` - by the time we get to this function we're already defaulted it to `true`/`false` without checking for a value in the repo config. Might be best to not have a value set for this in `defaultOptions` and do a check like:

```suggestion
    return this.options.autoMigrate || autoMigrateConfig || true
```

...although you can probably skip fetching `autoMigrateConfig` from the repo if `this.options.autoMigrate` is `true` or `false`.

**Can you please ensure there are tests to assert this behaviour is working as expected?**